Templated static member variables are initialized in the including module instead of the global module

When I include a header that contains a templated class with static member variables (and those variables are initialized outside the class) in the global module fragment of my module class, Clang determines that the initialization is in **my** module (not the GMF). See the following example: https://gcc.godbolt.org/z/ej6K4h6ds. Also pasting below for convenience.

`bar.h`:
```
template <typename T>
struct Singleton {
    static T* instance_;
    static T* get() {
        static bool init = false;
        if (!init) {
            init = true;
            instance_ = ::new T();
        }
        return instance_;
    }
};

template <typename T>
T* Singleton<T>::instance_ = nullptr;

struct s{};
inline void* foo() {
    return Singleton<s>::get();
}
```

`foo.ixx`:
```
module;
#include "bar.h"
export module foo;
```

This fails to compile because `declaration of 'instance_' in module foo follows declaration in the global module`, but the include is in the global module fragment so shouldn't the declaration also be there (not in `Foo`)? This compiles successfully on GCC.
```
bar.h:20:18: error: declaration of 'instance_' in module foo follows declaration in the global module
   20 | T* Singleton<T>::instance_ = nullptr;
      |                  ^
bar.h:13:13: note: in instantiation of static data member 'Singleton<s>::instance_' requested here
   13 |             instance_ = ::new T();
      |             ^
bar.h:25:26: note: in instantiation of member function 'Singleton<s>::get' requested here
   25 |     return Singleton<s>::get();
      |                          ^
bar.h:8:15: note: previous declaration is here
    8 |     static T* instance_;
```

--- a/clang/lib/Sema/Sema.cpp
+++ b/clang/lib/Sema/Sema.cpp
@@ -1104,9 +1104,13 @@ void Sema::ActOnStartOfTranslationUnit() {
 }
 
 void Sema::ActOnEndOfTranslationUnitFragment(TUFragmentKind Kind) {
-  // No explicit actions are required at the end of the global module fragment.
-  if (Kind == TUFragmentKind::Global)
+  if (Kind == TUFragmentKind::Global) {
+    // Perform Pending Instantiations at the end of global module fragment so
+    // that the module ownership of TU-level decls won't get messed.
+    llvm::TimeTraceScope TimeScope("PerformPendingInstantiations");
+    PerformPendingInstantiations();
     return;
+  }
 
   // Transfer late parsed template instantiations over to the pending template
   // instantiation list. During normal compilation, the late template parser

--- a/clang/test/Modules/pr125999.cppm
+++ b/clang/test/Modules/pr125999.cppm
@@ -1,0 +1,33 @@
+// RUN: rm -rf %t
+// RUN: mkdir -p %t
+// RUN: split-file %s %t
+//
+// RUN: %clang_cc1 -std=c++20 %t/foo.cppm -verify -fsyntax-only
+
+//--- bar.h
+template <typename T>
+struct Singleton {
+    static T* instance_;
+    static T* get() {
+        static bool init = false;
+        if (!init) {
+            init = true;
+            instance_ = ::new T();
+        }
+        return instance_;
+    }
+};
+
+template <typename T>
+T* Singleton<T>::instance_ = nullptr;
+
+struct s{};
+inline void* foo() {
+    return Singleton<s>::get();
+}
+
+//--- foo.cppm
+// expected-no-diagnostics
+module;
+#include "bar.h"
+export module foo;

